Sources/_MatchingEngine/Regex/Parse/Parse.swift

```diff
@@ -127,7 +127,7 @@ extension Parser {
       }
       fatalError("Unhandled termination condition")
     }
-    return ast
+    return .init(ast)
```

Comment:
Where do global options come in?

Reply:
We'll parse them here and store them on `AST`

Reply:
Oh, right, that will be based on top of this. SGTM